src/lpf2hub.ts

```diff
@@ -608,5 +539,187 @@ export class LPF2Hub extends Hub {
+                commons.distance = 5;
+                commons.partial = 7;
+                break;
+            }
+        }
+
+        const color = commons.color ? data[commons.color] : undefined;
+        const partial = commons.partial ? data[commons.partial] : undefined;
+        const distance = commons.distance ? toDistance(data[commons.distance], partial) : undefined;
+
+        if (color !== undefined) {
+            /**
+             * Emits when a color sensor is activated.
+             * @event LPF2Hub#color
+             * @param {string} port
+             * @param {Color} color
+             */
+            this.emit("color", port.id, color);
+        }
+
+        if (distance !== undefined) {
+            /**
+             * Emits when a distance sensor is activated.
+             * @event LPF2Hub#distance
+             * @param {string} port
+             * @param {number} distance Distance, in millimeters.
+             */
+            this.emit("distance", port.id, distance);
+        }
+
+        if (color !== undefined && distance !== undefined) {
+            /**
+             * A combined color and distance event, emits when the sensor is activated.
+             * @event LPF2Hub#colorAndDistance
+             * @param {string} port
+             * @param {Color} color
+             * @param {number} distance Distance, in millimeters.
+             */
+            this.emit("colorAndDistance", port.id, color, distance);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.COUNT) {
+            /**
+             * Emits when a count mode distance sensor is activated.
+             * @event LPF2Hub#count
+             * @param {string} port
+             * @param {number} count
+             */
+            this.emit("count", port.id, data[4]);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.REFLT) {
+            /**
+             * Emits when a reflect mode color sensor is activated.
+             * @event LPF2Hub#reflectivity
+             * @param {string} port
+             * @param {number} reflectivity
+             */
+            this.emit("reflectivity", port.id, data[4]);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.AMBI) {
+            /**
+             * Emits when a ambiant mode color sensor is activated.
+             * @event LPF2Hub#luminosity
+             * @param {string} port
+             * @param {number} luminosity
+             */
+            this.emit("luminosity", port.id, data[4]);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.RGB_I) {
+            /**
+             * Emits when a RGB mode color sensor is activated.
+             * @event LPF2Hub#rgb
+             * @param {string} port
+             * @param {number} r
+             * @param {number} g
+             * @param {number} b
+             */
+            this.emit("rgb", port.id, data[4], data[6], data[8]);
```

Comment:
I did not verify it myself by according to port info this mode should return 3 16-bit values in range 0-1023 (so it should be read with `readInt16LE`).
```
90842b0e7cad Port 01, mode 6, name RGB I
90842b0e7cad Port 01, mode 6, RAW min 0, max 1023
90842b0e7cad Port 01, mode 6, PCT min 0, max 100
90842b0e7cad Port 01, mode 6, SI min 0, max 1023
90842b0e7cad Port 01, mode 6, SI symbol RAW
90842b0e7cad Port 01, mode 6, Value 3 x 16bit, Decimal format 5.0
```